internal/services/cosmos/common/autoscale_settings.go

```diff
@@ -25,6 +26,23 @@ func ExpandCosmosDbAutoscaleSettings(d *pluginsdk.ResourceData) *documentdb.Auto
 	return &autoscaleSettings
 }
 
+func ExpandCosmosDbAutoscaleSettingsForGremlinAndSqlContainer(d *pluginsdk.ResourceData) *cosmosdb.AutoScaleSettings {
```

Comment:
This isn't for gremlin and SQL only? its for the new SDK? so can we keep the `ExpandCosmosDbAutoscaleSettingsResource` name and then rename the old one to `ExpandCosmosDbAutoscaleSettingsResourceLegacy` or `ExpandCosmosDbAutoscaleSettingsResourceTrack1` ?

this applies to all the new flatten and expand functions

Reply:
Sure. Updated.